components/sidebar_right/sidebar_right.tsx

```diff
@@ -14,40 +13,51 @@ import RhsThread from 'components/rhs_thread';
 import RhsCard from 'components/rhs_card';
 import ChannelInfoRhs from 'components/channel_info_rhs';
 import ChannelMembersRhs from 'components/channel_members_rhs';
-import Search from 'components/search/index.tsx';
+import Search from 'components/search/index';
 
 import RhsPlugin from 'plugins/rhs_plugin';
 
-export default class SidebarRight extends React.PureComponent {
-    static propTypes = {
-        isExpanded: PropTypes.bool.isRequired,
-        isOpen: PropTypes.bool.isRequired,
-        channel: PropTypes.object,
-        postRightVisible: PropTypes.bool,
-        postCardVisible: PropTypes.bool,
-        searchVisible: PropTypes.bool,
-        isPinnedPosts: PropTypes.bool,
-        isChannelFiles: PropTypes.bool,
-        isChannelInfo: PropTypes.bool,
-        isChannelMembers: PropTypes.bool,
-        isPluginView: PropTypes.bool,
-        previousRhsState: PropTypes.string,
-        rhsChannel: PropTypes.object,
-        selectedPostId: PropTypes.string,
-        selectedPostCardId: PropTypes.string,
-        actions: PropTypes.shape({
-            setRhsExpanded: PropTypes.func.isRequired,
-            showPinnedPosts: PropTypes.func.isRequired,
-            openRHSSearch: PropTypes.func.isRequired,
-            closeRightHandSide: PropTypes.func.isRequired,
-            openAtPrevious: PropTypes.func.isRequired,
-            updateSearchTerms: PropTypes.func.isRequired,
-            showChannelFiles: PropTypes.func.isRequired,
-            showChannelInfo: PropTypes.func.isRequired,
-        }),
+import {Channel} from '@mattermost/types/channels';
+import {RhsState} from 'types/store/rhs';
+
+type Props = {
```

Comment:
not exactly sure what is the current approach to this, but I like to have the typings in the `index.ts` file and separate it into `OwnProps`, `StateProps` and `Actions`. Then import it into the component file and combine them to a `Props` type. 

```
type Props = OwnProps & StateProps & Actions;
```

WDYT @hmhealey ?

Reply:
Please let me know about decision. Based on the decision, I can use the suggested approach in future PRs